src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -738,7 +743,12 @@ private void doPostRequestToSrs(SourceStorageSourceRecordsClient srsClient, Vert
           return;
         }
         srsResponse.bodyHandler(buffer -> handleSrsResponse(srsClient, promise, buffer));
-      }, e->  retrySRSRequest(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, retryAttempts, promise, 0, null));
+      }, e-> {
+        if (!retryTriggered.get()) {
```

Comment:
Do it atimically, otherwise you have race condition

Reply:
Thanks, done